### ISSUE TITLE
Check / integrate GovCert file extension list

### ISSUE THREAD

**Issue by theseion** [coreruleset/coreruleset#3143]
GovCert Switzerland has published a list of file extensions they think should be blocked: https://www.govcert.ch/downloads/blocked-filetypes.txt

**Comment by dune73** [comment]
List is described here: https://www.ncsc.admin.ch/ncsc/en/home/aktuell/news/news-archiv/update-ransomware-neue-vorgehensweise.html

It's targeted at email attachments, actually.

**Comment by dune73** [comment]
Looking over a body of 450K requests, I see the following file extensions from the list above requested by attackers, that CRS v4-dev did not find suspicious:

```
* .7z
* .application
* .ashx
* .asp (but we apparently can not block this)
* .bz2
* .bzip
* .exe
* .gz
* .pl
* .sh
```

The list has some merit, I think.

**Comment by fzipi** [comment]
@dune73 @theseion Should we follow up on this one?

**Comment by dune73** [comment]
Yes, I think we should add most of the entries to our list of restricted file extensions.

**Comment by RedXanadu** [comment]
To add some more context: we debated something similar last year and eventually decided against forbidding many of those archive-like file formats by default. IIRC, on the grounds that there's nothing inherently wrong with serving archive files for download and we didn't want to block by default, so it was added as an optional list.

_Ref:_ https://github.com/coreruleset/coreruleset/commit/6834797c4d97a47aa80f38c94ee4cb2bc15463af

**Comment by dune73** [comment]
Correct. Thanks for pointing this out @RedXanadu. It's a question of balancing and given this has been discussed, let's not reiterate this and leave it at this for the time being.

**Comment by theseion** [comment]
So, can we close the issue?

**Comment by Xhoenix** [comment]
Anything else needs to be done here, or we close the issue?

**Comment by Xhoenix** [comment]
I checked the [file](https://www.ncsc.admin.ch/dam/ncsc/de/dokumente/infos-it-spezialisten/govcert/govcert-ncsc_blocked-filetypes.txt.download.txt/govcert-ncsc_blocked-filetypes.txt), and a lot of extensions are not archive file extensions. I think it's safe to add those other extensions. Should I send a PR?

**Comment by theseion** [comment]
@Xhoenix yes, please do. It will make it easier to discuss once we see what the actual change would be.

**Comment by Xhoenix** [comment]
Looks like a lot of work to do. Going to add... 😄

**Comment by Xhoenix** [comment]
Created [#4068](https://github.com/coreruleset/coreruleset/pull/4068), and had only added "Executables" and "System Files" from [that list](https://www.ncsc.admin.ch/dam/ncsc/de/dokumente/infos-it-spezialisten/govcert/govcert-ncsc_blocked-filetypes.txt.download.txt/govcert-ncsc_blocked-filetypes.txt)
